features/telemetry/common/src/main/java/org/opennms/netmgt/telemetry/common/ipc/TelemetrySinkModule.java

```diff
@@ -171,7 +173,11 @@ public boolean isBlockWhenFull() {
 
     @Override
     public Optional<String> getRoutingKey(final TelemetryProtos.TelemetryMessageLog message) {
-        return Optional.of(String.format("%s@%s:%d", message.getLocation(), message.getSourceAddress(), message.getSourcePort()));
+        if (message.getUseRoutingKey()) {
+            return Optional.of(String.format("%s@%s:%d", message.getLocation(), message.getSourceAddress(), message.getSourcePort()));
+        } else {
+            return null;
```

Comment:
This is an `Optional` value and thus should not return `null`. Use `Optional.of()` and `Optional.empty()` instead.

Reply:
Done